xhu.py

```diff
@@ -135,6 +179,11 @@ def put_file(path):
     )
 
     dest_path.parent.mkdir(parents=True, exist_ok=True, mode=0o770)
+
+    quota = flask.request.args.get("q", "")
```

Comment:
This allows a client to provide an arbitrary quota, or leave out the ``q`` entirely. The quota needs to be part of the HMAC instead.

Reply:
Yes, very good point.